packages/auth/src/providers/cognito/tokenProvider/TokenStore.ts

```diff
@@ -145,8 +145,8 @@ export class DefaultTokenStore implements AuthTokenStore {
 		]);
 	}
 
-	async getDeviceMetadata(): Promise<DeviceMetadata | null> {
-		const authKeys = await this.getAuthKeys();
+	async getDeviceMetadata(username?: string): Promise<DeviceMetadata | null> {
+		const authKeys = await this.getAuthKeys(username);
```

Comment:
I would call `getAuthKeys` for encode an non encode `username` just in case it has non encoded value  (and use first the encodeand then if nothing is there try non-encode `username`

Reply:
updated